Documentation/rfc/v3api.proto

```diff
@@ -32,24 +43,11 @@ service KV {
   // LeaseRevoke revokes a lease. All the key attached to the lease will be expired and deleted.
   rpc LeaseRevoke(LeaseRevokeRequest) returns (LeaseRevokeResponse) {}
 
-  // LeaseAttach attaches keys with a lease.
-  rpc LeaseAttach(LeaseAttachRequest) returns (LeaseAttachResponse) {}
```

Comment:
> Will Put supersede LeaseAttach's functionality because of the KV lease field?

Yes.

>  If so, would this mean that attaching a lease would be a read-modify-write (e.g., I must know the key's value before I can do the PUT to attach the lease)?

One nice thing about proto is that we can know if one field is set or not. So we can do "read-modify-write" at server side. So the client does not know that. If they only want to change lease, they just need to leave value unset. 

But we probably will not do rmw for now. We will see the needs and can add it.